### ISSUE TITLE
ignore fields specifically 

### ISSUE THREAD

**Issue by jfragosoperez** [bluelinelabs/LoganSquare#38]
Hi, at first congratulations, you've done a great job with the lib, the performance is astonishing. 

I was wondering if there is a way to ignore a field specifically for either serializing/parsing, instead of ignoring for both operations.

It is quite common in APIs to have a service that is sent and the server answers with an object of the same type. For instance, a service that updates the profile of a user normally sends a user object with some fields filled and receives as response an updated user. Moreover, sometimes the response can have other fields that we do not want to serialize and although the server check only the needed fields, I find a bad practice to send parameters that wouldn't need to be sent.

Another example could be having a BaseApiResponse super class that acts as a wrapper (contains the response) and other fields like for instance those ones related to error handling. It is something usual that those fields are repeated in most of the services, but those fields shouldn't be sent when sending an instance of the object to the server.

Maybe there is just a way to handle this, I have had a look and I have not found a way to do this, but just in case I am missing something, please excuse me.

I know that a simple way to handle this is to remove chars from the inputstream before sending, but I don't like this approach, I think that this should be done before.

1- I think that a good way to handle this should be using a kind of annotation that lets you exclude a field to be taken into account when serializing or parsing, for instance "@IgnoreSerialization" and "@IgnoreParsing" or just a single one like "@IgnoreSpecifically(op = "serialization").

2- Another way would be to have an annotation like those ones you are using when operations have finished (i.e. "@OnJsonParseComplete") but that is taken into account in the process of the operation. By this way returning a kind of facade have a method ignoreField(field) and this removes unwanted fields.

I would like to know your thoughts, maybe you think that this could affect a little bit the performance and you discard an option like that, but first option would not damage so much performance. 

As I told maybe there is just a way to do this, if there is, many apologies. 

Thanks a lot

**Comment by EricKuck** [comment]
Thanks for all the detail on this! I think this is a great idea, and I'll be adding support for it in the next version, which should be sometime early next week. I'm planning to add an optional argument to the current `@JsonIgnore` annotation called `ignoreFor` that can be `PARSE_AND_SERIALIZE` (the default), `PARSE_ONLY`, or `SERIALIZE_ONLY`. Sound good?

**Comment by jfragosoperez** [comment]
Sounds great. If you need some help, let me know and I will help to develop this feature.

**Comment by EricKuck** [comment]
Already started :)

**Comment by EricKuck** [comment]
Alright, this is done in commit 3f120c2012fdf75ed142d9fdfdc2b62ca670a4d5. It's not live on JCenter or Maven Central yet, so you'll have to build from source if you want to use it ASAP. I'll probably push it on Friday or so.

**Comment by jfragosoperez** [comment]
This is what is called being faster than a lightning bolt! haha Thanks a lot :) no worries, I think that I can wait for the release, else I will clone the repo ;) Thanks again

**Comment by jfragosoperez** [comment]
Hi Eric,
Just wanted to know if there are any news on this update. I also wanted to know if you would consider an addition to skip null values for object serialization. It is the default behaviour for Gson and can also be customized using Jackson:
```
mapper.setSerializationInclusion(Include.NON_NULL)
```
I think that this could be a good addition, but I wanted to know if you are happy with this feature before developing it and pushing a PR. If you think so, we can label this as a new issue.

Thank you a lot
